meetings/2021-07/july-14.md

```diff
@@ -0,0 +1,1028 @@
+
+SYG: Has it been discussed that we reserve one for JavaScript? Because otherwise, it means you have to always use different syntax to import JS.
+
+DDC: I mean that would be another alternative that I think could allow, like, it could be say allow asserts type equals JavaScript's. That that's basically the same thing as not having the type of solution and then the rest of the set of reserved strings is banned and causes a failure. I think that would also be reasonable. We'd also I think at some point has been discussed that like, like we discussed whether or not a "type JavaScript" assertion should be supported and I think we'd elected not to because it's better not have two ways to do the same thing, but I think it would be reasonable to allow one such assertion. It's just in any case not - I think that's almost a separate question like whether something like, like, this should be allowed is like one question of whether that's a type of solution that should work in say the web and then there's kind of a separate question of whether we should have a block list of all the types that aren't that, or like all the types including that.
+
+SYG: Yeah, I think I agree that these are orthogonal concerns.
+
+JHD: I don't think there's any way we can phrase this that won't be relatively trivial to bypass. If we have a list, then someone will come up with a new way to mean JavaScript, right? Like JS, JavaScript, jscript. I don't know. Someone will come up with something, but then like, if it's if we do the like prose option, I mean Source, Testament, text module records, I can make my own kind of module record as a host that is a source text module record but with some optional trivial difference, which means effectively, includes all sorts text module records and then I can bypass this assertion entirely and still be in compliance. So, like, I think that on some level, I don't think any host is likely to do this, and if they do they wouldn't be able to leverage almost all the code out there that's using no assertions to bring in JavaScript. So like it kind of seems - it seems very unlikely to me that without a restriction this would become an interop concern because I can't see enough people using such a host with that assertion. So it kind of seems like the way things are is ideal.
```

Comment:
```suggestion
JHD: I don't think there's any way we can phrase this that won't be relatively trivial to bypass. If we have a list, then someone will come up with a new way to mean JavaScript, right? Like JS, JavaScript, jscript. I don't know. Someone will come up with something, but then like, if it's if we do the like prose option, I mean Source Text Module Records, I can make my own kind of module record as a host that is a source text module record but with some optional trivial difference, which means effectively, includes all sorts text module records and then I can bypass this assertion entirely and still be in compliance. So, like, I think that on some level, I don't think any host is likely to do this, and if they do they wouldn't be able to leverage almost all the code out there that's using no assertions to bring in JavaScript. So like it kind of seems - it seems very unlikely to me that without a restriction this would become an interop concern because I can't see enough people using such a host with that assertion. So it kind of seems like the way things are is ideal.
```